packages/xo-web/src/xo-app/xostor/index.js

```diff
@@ -0,0 +1,559 @@
+          predicate={state.networksPredicate}
+          value={state.networkId}
+        />
+      </CardBlock>
+    </Card>
+  ),
+])
+
+const DisksCard = decorate([
+  provideState({
+    initialState: () => ({
+      hostId: undefined,
+      onlyShowXostorDisks: true,
+    }),
+    effects: {
+      toggleState,
+      onHostChange(_, host) {
+        this.state.hostId = host?.id
+      },
+      _onDiskChange(_, disk) {
+        this.effects.onDiskChange(disk, this.state.hostId)
+      },
+    },
+    computed: {
+      _blockdevices: async state =>
+        state.isHostSelected && state.isXcpngPool ? (await getBlockdevices(state.hostId)).blockdevices : undefined,
+      _disks: state =>
+        state.onlyShowXostorDisks ? state._blockdevices?.filter(xostorDiskPredicate) : state._blockdevices,
+      predicate: state => host => host.$pool === state.poolId,
+      isHostSelected: state => state.hostId !== undefined,
+      selectableDisks: state =>
+        state._disks
+          ?.filter(disk => !state.disksByHost[state.hostId]?.some(_disk => _disk.name === disk.name))
+          .sort((prev, next) => Number(next.size) - Number(prev.size)),
+    },
+  }),
+  injectState,
+  ({ effects, state }) => (
+    <Card>
+      <CardHeader>{_('disks')}</CardHeader>
+      <CardBlock>
+        <Row>
+          <Col size={8}>
+            <Row>
+              <div className={styles.disksSelectors}>
+                <Col size={6}>
+                  <SelectHost
+                    disabled={!state.isPoolSelected}
+                    onChange={effects.onHostChange}
+                    predicate={state.predicate}
+                    value={state.hostId}
+                  />
+                </Col>
+                <Col size={6}>
+                  <label>
+                    <input
+                      checked={state.onlyShowXostorDisks}
+                      onChange={effects.toggleState}
+                      name='onlyShowXostorDisks'
+                      type='checkbox'
+                    />{' '}
+                    {_('onlyShowXostorRequirements', { type: _('disks') })}
+                  </label>
+                  {state.isPoolSelected && !state.isXcpngPool && (
+                    <p className='text-danger mb-0'>
+                      <Icon icon='alarm' /> {_('cantFetchDisksFromNonXcpngHost')}
+                    </p>
+                  )}
+                  <Select
+                    disabled={!state.isHostSelected || !state.isPoolSelected || !state.isXcpngPool}
+                    onChange={effects._onDiskChange}
+                    optionRenderer={rendererDiskSelect}
+                    options={state.isHostSelected ? state.selectableDisks : []}
+                    placeholder={_('selectDisks')}
+                    value={null}
+                  />
+                </Col>
+              </div>
+            </Row>
+            <Row>
+              <Col>
+                <ListSelectedDisks hostId={state.hostId} />
```

Comment:
Is it really necessary to show the list of disks here again since the lists on the right are uncollapsed by default anyway? Or do you think it would be hard to see which host is adding a disk from? Maybe we could uncollapse the right disk on the right side when the user selects a disk? I'm not sure what's better.

Reply:
For me, you should see the list on the right as an overview. (that's why it's uncollapsed by default).
In case we remove this list and use the right list to manage disks, if there are 7 hosts, with multiple disks on each, it will force users to scroll to see/manage their disks.